activesupport/lib/active_support/cache/file_store.rb

```diff
@@ -21,6 +21,15 @@ class FileStore < Store
       GITKEEP_FILES = [".gitkeep", ".keep"].freeze
 
       def initialize(cache_path, options = nil)
```

Comment:
How about changing the method signature instead?

```suggestion
      def initialize(cache_path, **options)
```

Reply:
@kamipo the problem is with the `cache_path` argument, not with `options`. Even with that change, `cache_path` would still be `{}` if you do `config.cache_store = :file_store` with no path.

Reply:
The reason to be passed the exracted `**options` to `cache_path` is that the `initialize` doesn't accept kwargs, if `initialize` allows kwargs, the exracted `**options` isn't passed to `cache_path`.

https://github.com/rails/rails/blob/291a3d2ef29a3842d1156ada7526f4ee60dd2b59/activesupport/lib/active_support/cache.rb#L62-L66

Reply:
hm... looks like that is true only for Ruby 3.0, the extracted `**options` is still passed to `cache_path` if it is required in Ruby 2.7.

Reply:
Hmmm, I tried your proposed change but the test I added fails with it (no error raised). And the `cache_path` is still `{}`.

```ruby
  def test_file_store_requires_a_path
    assert_raises(ArgumentError) do
      store = ActiveSupport::Cache.lookup_store :file_store
      puts store.cache_path
    end
  end
```

```
{}
F

Failure:
CacheStoreSettingTest#test_file_store_requires_a_path [/Users/alex/code/rails/activesupport/test/cache/cache_store_setting_test.rb:25]:
ArgumentError expected but nothing was raised.
```

Reply:
I think that's because `*parameters` is always passed as the first argument. `FileStore` is the only cache store that expects as string as its first (and only) argument; all other cache stores expect a hash (eg. `MemCacheStore`) or nothing (eg `NullStore`). So we need to jump through some hoops in file store to validate the arguments.

Reply:
Actually it is a regression for 9845cd6338e44d9fff39ea1cd64851af085133ac, the `test_file_store_requires_a_path` test is passed before 9845cd6338e44d9fff39ea1cd64851af085133ac.

I think that the extracted `options` should not be passed if that is empty, we can clean it up once Ruby 2.7 support is dropped.

Reply:
Ahhh, good find. I updated based on that, can you please have another look? Thanks for your help @kamipo